extensions/antora/xref-hash-validator/xref-hash-validator.js

```diff
@@ -0,0 +1,142 @@
+                    return [id]
+                    
+                }
+                return []
+            })
+            
+            // any anchor element in the HTML that has an href attribute can originate from an xref in the source file for this HTML page
+            const anchors = parseHTML(file.contents.toString()).querySelectorAll('a')
+            file.xrefChecker.internalLinks = anchors.flatMap((a) => {
+                // ignore unresolved links that Antora has already logged
+                if (a.classList.contains('unresolved')) return []
+
+                const href = a.getAttribute('href')
+
+                if (!href) return []
+
+                // if an id on this page matches this href, we can ignore it - it is essentially self-verifying
+                if (file.xrefChecker.linkTargets.includes(href.replace(/^#/, ''))) return []
+
+                if (href && !href.startsWith('http') && !href.startsWith('/docs')) {
+                    return [href]
+                }
+                return []
+            })
+
+        })
+        
+        // we can go back into the contentCatalog and check the xref data we just added
+        // for every file in the contentCatalog, check if the xrefChecker.internalLinks are valid
+        // by verifying that for an internal link to a section, the hash does actually exist in the target page
+        files.forEach( (file) => {
+            if (!file.out || !file.asciidoc) return
+    
+            // what is the full url of each target?
+            file.xrefChecker.internalLinks.forEach((link) => {
+
+                searchForThisURL = new URL(path.join(file.pub.url, link), playbook.site.url)
+
+                // if there's no hash the xref is already checked by Antora
+                // this check shouldn't ever match anyway because we should have already filtered out links without hashes
+                if (!searchForThisURL.hash) return
+                
+                // Find the file in the contentcatalog that has a file.pub.url value that matches the pathname of the link target
+                // where the pathname is a relative path from the site root, which is also what pub.url represents
+                const targetFile = contentCatalog
+                .findBy({ family: 'page' })
+                .filter((page) => page.pub)
+                .filter((f) => {
+                    return f.pub.url === searchForThisURL.pathname
+                })[0]
+
+                const hashTarget = decodeURI(searchForThisURL.hash.replace('#', ''))
+
+                // if the anchor doesn't exist anywhere...
+                if (!linkTargets[hashTarget]) {
+
+                    if (linkTargets[hashTarget.replace(' ', '-')]) {
+                        logger[logLevel]({ file: file.src, source: file.src.origin }, 'anchor %s not found in target page %s - did you mean %s?', decodeURI(searchForThisURL.hash), targetFile.src.path, hashTarget.replace(' ', '-'))
```

Comment:
```suggestion
                        logger[logLevel]({ file: file.src, source: file.src.origin }, 'anchor %s not found in target page %s - did you mean %s ?', decodeURI(searchForThisURL.hash), targetFile.src.path, hashTarget.replace(' ', '-'))
```
I think this will make copy-paste work easier in terminal. At least my terminal will correctly select the suggested `%s` is surrounded by spaces, whereas it will pick up the trailing `?` otherwise.